sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -140,14 +141,18 @@ object V2ScanRelationPushDown extends Rule[LogicalPlan] with PredicateHelper wit
                         val count = aggregate.Count(avg.child).toAggregateExpression(isDistinct)
                         // Closely follow `Average.evaluateExpression`
```

Comment:
Now I have a better idea: how about we just take `Average.evaluateExpression` and replace its sum/count attribute with new ones?
```
avg.evaluateExpression transform {
  case a: Attribute if a.semanticEquals(avg.sum) => sum
  case a: Attribute if a.semanticEquals(avg.count) => count
}
```

Reply:
Let me have a try.